prestoadmin/server.py

```diff
@@ -353,18 +353,33 @@ def check_server_status(client):
     Returns:
         True or False
     """
-    result = True
     time = 0
+
+    if len(get_coordinator_role()) < 1:
+        warn('No coordinator defined.  Cannot verify server status.')
+    client = PrestoClient(get_coordinator_role()[0], env.user)
+    node_id = lookup_string_config('node.id', os.path.join(constants.REMOTE_CONF_DIR, 'node.properties'), env.host)
     while time < RETRY_TIMEOUT:
-        result = client.execute_query(SYSTEM_RUNTIME_NODES)
-        if not result:
+        try:
+            client.execute_query(SYSTEM_RUNTIME_NODES)
+        except ConfigurationError as e:
+            _LOGGER.warn(e)
+            break
+        if not _is_in_rows(node_id, client.get_rows()):
             run('sleep %d' % SLEEP_INTERVAL)
             _LOGGER.debug('Status retrieval for the server failed after '
                           'waiting for %d seconds. Retrying...' % time)
             time += SLEEP_INTERVAL
         else:
-            break
-    return result
+            return True
+    return False
+
+
+def _is_in_rows(value, rows):
+    for row in rows:
+        if value in row:
```

Comment:
On the one hand, this doesn't depend on the current table layout, on the other, it's a little gross. I guess I'm OK with it though because of the layout independence.